pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -784,6 +784,8 @@ func translateHeadersOperations(headers *networking.Headers) headersOperations {
 	}
 }
 
+const singleDNSLabelWildcardRegex = "^[-a-zA-Z]*"
```

Comment:
What is this used for ? DNS label includes digits too AFAIK.

Reply:
Good point, let me find a more robust regex.

Basically we want to match `*.example.com` to `foo.example.com` but not `foo.bar.example.com`. Like TLS verification matching, its a 1 label wildcard.

Envoy doesn't have a native way to d othis